BlueprintUI/Sources/Layout/GridRow.swift

```diff
@@ -0,0 +1,244 @@
+///   row.add(child: .proportional(name, 0.75))
+///   row.add(child: .proportional(number, 0.25))
+///   row.add(child: .absolute(status, 50))
+/// }
+/// ```
+///
+/// ## Expected layout:
+///
+/// ```
+/// ┌────────────────────────────┬─┬────────┬─┬──────────────────┐
+/// │            name            │ │ number │ │      status      │
+/// │            (75%)           │8│  (25%) │8│     (100 pts)    │
+/// │                            │ │        │ │                  │
+/// ●──────────── 150 ───────────● ●── 50 ──● ●─────── 100 ──────●
+/// └────────────────────────────┴─┴────────┴─┴──────────────────┘
+/// ●──────────────────────────── 316 ───────────────────────────●
+/// ```
+public struct GridRow: Element {
+    // MARK: - properties -
+    /// How children are aligned. By default, `.fill`.
+    public var verticalAlignment: VerticalAlignment = .fill
+
+    /// The space between children. By default, 0.
+    public var spacing: CGFloat = 0
+
+    /// The child elements to be laid out. By default, an empty array.
+    public var children: [Child] = []
+
+    // MARK: - initialization -
+    public init(configure: (inout GridRow) -> Void = { _ in }) {
+        configure(&self)
+    }
+
+    // MARK: - mutations -
+    public mutating func add(child: Child) {
+        children.append(child)
+    }
+
+    // MARK: - GridRow+Element -
+    public var content: ElementContent {
+        ElementContent(layout: GridRowLayout(verticalAlignment: verticalAlignment, spacing: spacing)) { builder in
+            children.forEach {
+                builder.add(traits: $0.width, key: $0.key, element: $0.element)
+            }
+        }
+    }
+
+    public func backingViewDescription(bounds: CGRect, subtreeExtent: CGRect?) -> ViewDescription? {
+        nil
+    }
+}
+
+// MARK: - child modeling -
+public extension GridRow {
+    enum VerticalAlignment: Equatable {
+        /// Stretch children to fill the row vertically. In other words, size children to share the height of the
+        /// tallest child.
+        case fill
+        /// Align children to the top of the row.
+        case top
+        /// Align children in the center of the row.
+        case center
+        /// Align children to the bottom of the row.
+        case bottom
+    }
+
+
+    /// A child of a `GridRow`.
+    struct Child {
+        // MARK: - properties -
+        /// The element displayed in the `Grid`.
+        public var element: Element
+        /// A unique identifier for the child.
+        public var key: AnyHashable?
+        // The sizing for the element.
+        public var width: Width
+
+        // MARK: - initialialization -
+        public init(element: Element, key: AnyHashable? = nil, width: Width) {
+            self.element = element
+            self.key = key
+            self.width = width
+        }
+
+        // MARK: - convenience initialization -
+        public static func absolute(_ element: Element, _ width: CGFloat) -> Child {
```

Comment:
Nit: I would make the `element` the last parameter here, so the syntax of creating elements inline is cleaner. This is generally the style we try to follow in blueprint. Eg:

```
.absolute(50, Label(text: "...") {
   $0.font = ...
   $0.alignment = ...
})
```

Vs

```
.absolute( Label(text: "...") {
   $0.font = ...
   $0.alignment = ...
}, 50)
```

Reply:
Great point. I'll update, thanks.